Sources/NIO/CircularBuffer.swift

```diff
@@ -126,29 +136,42 @@ extension CircularBuffer {
         assert(newCapacity % 2 == 0)
 
         newBacking.reserveCapacity(newCapacity)
-        newBacking.append(contentsOf: self.buffer[self.headIdx..<self.buffer.count])
-        if self.headIdx > 0 {
-            newBacking.append(contentsOf: self.buffer[0..<self.headIdx])
+        newBacking.append(contentsOf: self.buffer[self.headIdx._value..<self.buffer.count])
+        if self.headIdx._value > 0 {
+            newBacking.append(contentsOf: self.buffer[0..<self.headIdx._value])
         }
-        newBacking.append(contentsOf: repeatElement(nil, count: newCapacity - newBacking.count))
-        self.tailIdx = self.buffer.count
-        self.headIdx = 0
+        let repeatitionCount = newCapacity - newBacking.count
+        newBacking.append(contentsOf: repeatElement(nil, count: repeatitionCount))
+        self.headIdx = Index(value: 0, headValue: 0, underlyingCount: newBacking.count)
+        self.tailIdx = Index(value: newBacking.count - repeatitionCount, headValue: self.headIdx._value, underlyingCount: newBacking.count)
         self.buffer = newBacking
     }
+    
+    /// Return element `offset` from first element.
+    ///
+    /// *O(1)*
+    public subscript(offset: Int) -> Element {
```

Comment:
would you mind making this label named, ie `buffer[offset: 7]` which would mean that it would be `public subscript(offset offset: Int) -> Element`